ktor-io/common/src/io/ktor/utils/io/ByteChannel.kt

```diff
@@ -243,4 +269,8 @@ public class ByteChannel(public val autoFlush: Boolean = false) : ByteReadChanne
 /**
  * Thrown when a coroutine awaiting I/O is replaced by another.
  */
-public class ConcurrentIOException(taskName: String) : IllegalStateException("Concurrent $taskName attempts")
+public class ConcurrentIOException(
+    taskName: String,
+    cause: Throwable? = null
+) : IllegalStateException("Concurrent $taskName attempts", cause) {
```

Comment:
Should we provide a `@Deprecated(HIDDEN)` constructor with an old signature for binary compatibility purpose?

Reply:
yep, let me add